dotcom-rendering/src/web/components/PinnedPost.tsx

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import {
+	neutral,
+	space,
+	textSans,
+	news,
+	from,
+	visuallyHidden,
+	transitions,
+	focusHalo,
+	height,
+} from '@guardian/source-foundations';
+import {
+	SvgMinus,
+	SvgPlus,
+	SvgPinned,
+} from '@guardian/source-react-components';
+import { timeAgo } from '@guardian/libs';
+import { css } from '@emotion/react';
+
+const pinnedPostContainer = css`
+	border: 3px solid ${news[300]};
+	padding-bottom: ${space[1]}px;
+	margin-bottom: ${space[9]}px;
+	position: relative;
+	background: ${neutral[100]};
+	#pinned-post-checkbox:checked ~ #collapsible-body {
+		max-height: fit-content;
+	}
+	#pinned-post-checkbox:checked ~ #overlay,
+	#pinned-post-checkbox ~ label #svgminus,
+	#pinned-post-checkbox:checked ~ label #svgplus {
+		display: none;
+	}
+	#pinned-post-checkbox ~ label #svgplus,
+	#pinned-post-checkbox:checked ~ label #svgminus {
+		display: block;
+	}
+	#pinned-post-checkbox ~ label::after {
+		content: 'Show more';
+	}
+	#pinned-post-checkbox:checked ~ label::after {
+		content: 'Show less';
+	}
+`;
+
+const pinnedPostRow = css`
+	background: ${news[300]};
+	height: 2rem;
+	display: flex;
+	align-items: center;
+	svg {
+		fill: ${neutral[100]};
+		height: 2rem;
+	}
+`;
+
+const timeAgoText = css`
+	${textSans.small({ fontWeight: 'bold' })};
+	color: ${neutral[100]};
+	margin-left: 2.6rem;
+`;
+
+const overlay = css`
+	background-image: linear-gradient(
+		0deg,
+		${neutral[100]},
+		${neutral[100]} 40%,
+		rgba(255, 255, 255, 0)
+	);
+	height: 5rem;
+	position: absolute;
+	bottom: 0;
+	width: 100%;
+	display: block;
+`;
+
+const button = css`
+	display: inline-flex;
+	justify-content: space-between;
+	align-items: center;
+	box-sizing: border-box;
+	border: none;
+	cursor: pointer;
+	transition: ${transitions.medium};
+	text-decoration: none;
+	white-space: nowrap;
+	:disabled {
+		cursor: not-allowed;
+	}
+	&:focus {
+		${focusHalo};
+	}
+	background: ${news[300]};
+	margin-left: 0.625rem;
+	position: absolute;
+	bottom: -1.5rem;
+	${textSans.medium({ fontWeight: 'bold' })};
+	height: ${height.ctaMedium}px;
+	min-height: ${height.ctaMedium}px;
+	padding: 0 ${space[5]}px;
+	border-radius: ${height.ctaMedium}px;
+	padding-bottom: 0.125rem;
+	color: white;
+	${from.tablet} {
+		margin-left: 3.75rem;
+	}
+`;
+
+const collapsibleBody = css`
+	max-height: 40vh;
+	overflow: hidden;
+`;
+
```

Comment:
I'm just dropping this comment here because I don't know where else it should go but do we need a bit more space above the button here?
<img width="742" alt="Screenshot 2022-03-17 at 16 26 26" src="https://user-images.githubusercontent.com/1336821/158848213-e3829f73-3736-4be7-a665-3fbca54ec0ee.png">

Reply:
Done